tests/hilti/codegen/init.hlt

```diff
@@ -0,0 +1,20 @@
+# @TEST-EXEC: hiltic -j %INPUT >>output
+# @TEST-EXEC: hiltic -j -o a.hlto %INPUT && hiltic -j a.hlto >>output
+# @TEST-EXEC: btest-diff output
+
+module Test {
+
+import hilti;
+
+init function void test_init() {
+    hilti::print("init");
+}
+
+preinit function void test_preinit() {
+    # Calling runtime functionality isn't well-defined from
+    # inside a preinit function but this happens to work.
+    hilti::print("preinit");
+}
```

Comment:
We support multiple `init` and `preinit` functions and should maybe test that here (if hooks of same linkage emit identical strings this shouldn't mess up the btest-diff test).